src/player.cpp

```diff
@@ -0,0 +1,63 @@
+#include "player.h"
+#include <iostream>
+
+// Helper function
+/**
+ * @brief Trim trailing whitespace from a string
+ * 
+ * @param s string to trim
+ */
+void rtrim(std::string& s) {
+  s.erase(s.find_last_not_of(" \n\r\t") + 1);
+} 
```

Comment:
extra nit
```suggestion
namespace {

void rtrim(std::string& s) {
  s.erase(s.find_last_not_of(" \n\r\t") + 1);
} 

};
```

Reply:
https://medium.com/@mane.tako/understanding-unnamed-namespaces-in-c-7d41d367fd47